addons/purchase_stock/models/stock_move.py

```diff
@@ -175,8 +175,8 @@ def _get_value_from_account_move(self, quantity, at_date=None):
             valuation_data['quantity'] = quantity
             valuation_data['value'] = quantity * value / aml_quantity
         account_moves = self.env['account.move.line'].browse(aml_ids).move_id
-        valuation_data['description'] = _('%(value)s for %(quantity)s %(unit)s from %(bills)s',
-            value=value, quantity=quantity, unit=self.product_id.uom_id.name,
+        valuation_data['description'] = self.env._('%(value)s for %(quantity)s %(unit)s from %(bills)s',
+            value=self.company_currency_id.format(value), quantity=aml_quantity, unit=self.product_id.uom_id.name,
             bills=', '.join(account_move.name for account_move in account_moves))
```

Comment:
If we want to give a clean description, I wonder if we could just use `account_moves.mapped('name')` here, as when an `Iterable` is given as parameter for a translated string, it will be formatted with `format_list`.
So instead of `name1, name2, name3` we'd have `name1, name2 and name3`